web3/_utils/method_formatters.py

```diff
@@ -485,8 +485,13 @@ def raise_solidity_error_on_revert(response: RPCResponse) -> RPCResponse:
     if not isinstance(response['error'], dict):
         raise ValueError('Error expected to be a dict')
 
-    # Parity/OpenEthereum case:
     data = response['error'].get('data', '')
+
+    # Ganache case:
+    if isinstance(data, dict) and response['error'].get('message'):
```

Comment:
I believe this will swallow the Geth case, so I think we need to move it below line 507.

Reply:
In Geth case type of the `data` is `str` if I'm not mistaken.  
And we should perform this check before `if data.startswith('Reverted '):` since it's exactly what causes errors on Ganache (see #1793).

Reply:
Ah, you're right 👍

Reply:
Test case(s) should be added to cover the different response types. @dmitry-tk if you have time for this and are comfortable doing it, great - if not, I can take over after I finish another todo.

Reply:
Your Ganache test looks good. Looks like we just need to add one for a Geth-specific payload.